packages/aws-cdk-lib/aws-ecs/lib/amis.ts

```diff
@@ -97,6 +97,13 @@ export interface EcsOptimizedAmiProps {
    * @default false
    */
   readonly cachedInContext?: boolean;
+  /**
+   * Kernel version used.
+   * Only supported for Amazon Linux 2.
+   *
+   * @default none, uses the recommended kernel version
+   */
+  readonly kernel?: ec2.AmazonLinux2Kernel;
```

Comment:
Discussed offline, this should not be undeprecated...sorry about that, I was wrong. This entire enum should be deprecated and replaced with the linked `EcsOptimizedImage` instead.

Reply:
@comcalvi I replied to your comment [here](https://github.com/aws/aws-cdk/pull/29809#discussion_r1644357838). Can you have a look?

>  This entire props should be deprecated and replaced with the linked EcsOptimizedImage instead.

`EcsOptimizedImage` is a class and its private constructor is still using the deprecated `EcsOptimizedAmiProps`. There are no other Props interfaces for `EcsOptimizedImage` constructor. 

Is it time to remove `@deprecated` already?

Reply:
It looks like we deprecated the props and decided to create a bunch of factory methods that take an `Options` instead of `props`. I think your original solution was actually correct; leave these as deprecated (since they should be marked internal, and if we remove the deprecation we'll never remember to fix this) and just add the new property.

@TheRealAmazonKendra is correct about this, in principle, since a deprecated marker implies that these props have been directly replaced by something else. That is not what is going on here. Instead, these props are supposed to be hidden away behind a set of factory methods. So, even if we had not missed these when we did v2, the optimal would still be what you've done here; add `kernel` to these props. 

Sorry for the churn on this one, please leave the `@deprecated` and add the new property to those deprecated props.

Reply:
@comcalvi Thanks for the update 🙏 
I am not entirely confident I am following what has been the final decision regarding how this change should be incorporated but I tried to revert to a previous state in this branch history; namely (c107fe0f).

Please let me know if I misunderstood your last comment. Thanks again